### PR TITLE
feature/SMHE-2158-Object-Config-for-Dlms-adapter-UpdateFirmware-and-DecoupleMBusDeviceCommandExecutor

### PR DESCRIPTION
**Pull request by harrymiddelburgetall**
Implemented ObjectConfig for UpdateFirmwareCommandExecutor and DecoupleMBusDeviceCommandExecutor

### CODE REVIEW

**Comment by harrymiddelburgetall** [comment]
✔ https://jenkins.fdp.osgp.cloud/job/gxf-smhe-cucumber-build/258/

**Review by natasja-n — DISMISSED** [review]

File: osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json [hunk 1]
```diff
@@ -3048,7 +3048,177 @@
...
+          "datatype": "array",
+          "valuetype": "DYNAMIC",
+          "access": "R"
+        }
+      ]
+    },
+    {
+      "tag": "MBUS_CLIENT_SETUP",
+      "description": "M-Bus client setup",
+      "note": null,
```
Comment (stefanermens) [inline osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json]: When note is null, this line can be removed.

File: osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json [hunk 2]
```diff
@@ -3048,7 +3048,177 @@
...
+      "class-id": 72,
+      "version": 0,
+      "obis": "0.x.24.1.0.255",
+      "group": "GAS",
+      "meterTypes": ["SP","PP"],
+      "properties": {},
```
Comment (stefanermens) [inline osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json]: When there are no properties, this line can be removed.

File: osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json [hunk 3]
```diff
@@ -3048,7 +3048,177 @@
...
+          "valuetype": "FIXED_IN_METER",
+          "value": "IDENTIFICATION_NUMBER",
+          "access": "RW"
+        },
+        {
+          "id": 7,
+          "description": "manufacturer id",
+          "datatype": "long-unsigned",
+          "valuetype": "FIXED_IN_METER",
+          "value": "MANUFACTURER_ID",
```
Comment (stefanermens) [inline osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json]: When valuetype is FIXED_IN_METER then value should not be specified.

File: osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json [hunk 4]
```diff
@@ -3048,7 +3048,177 @@
...
+          "valuetype": "FIXED_IN_PROFILE",
+          "value": "0",
+          "access": "RW"
+        },
+        {
+          "id": 6,
+          "description": "identification number",
+          "datatype": "double-long-unsigned",
+          "valuetype": "FIXED_IN_METER",
+          "value": "IDENTIFICATION_NUMBER",
```
Comment (stefanermens) [inline osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json]: When valuetype is FIXED_IN_METER then value should not be specified.

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/DeviceChannelsHelper.java [hunk 5]
```diff
@@ -110,28 +113,23 @@ protected void resetMBusClientAttributeValues(
...
-                    MbusClientAttribute.VERSION, UINT_8_ZERO, obiscode, classId))
+                    MbusClientAttribute.VERSION, UINT_8_ZERO, device, channel))
             .addExecutor(
                 this.getMbusAttributeExecutor(
-                    MbusClientAttribute.DEVICE_TYPE, UINT_8_ZERO, obiscode, classId))
+                    MbusClientAttribute.DEVICE_TYPE, UINT_8_ZERO, device, channel))
             .addExecutor(
                 this.getMbusAttributeExecutor(
-                    MbusClientAttribute.PRIMARY_ADDRESS, UINT_8_ZERO, obiscode, classId));
+                    MbusClientAttribute.PRIMARY_ADDRESS, UINT_8_ZERO, device, channel));
```
Comment (stefanermens) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/DeviceChannelsHelper.java]: The `getMbusAttributeExecutor` function will get the M-Bus client setup object from the object config every call again. Wouldn't it be better to get the object once and pass it to the function?

**Review by stefanermens — CHANGES_REQUESTED** [review]

File: osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json (hunk 1 above)
Comment (harrymiddelburgetall) [reply osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json]: This is commonly defined like this. Fixed this occurance now. Other occurances will be fixed  in other PR.

File: osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json (hunk 2 above)
Comment (harrymiddelburgetall) [reply osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json]: This is commonly defined like this. Fixed this occurance now. Other occurances will be fixed  in other PR.

File: osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json (hunk 3 above)
Comment (harrymiddelburgetall) [reply osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json]: Removed the value for this attribute in all protocol versions

File: osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json (hunk 4 above)
Comment (harrymiddelburgetall) [reply osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json]: Removed the value for this attribute in all protocol versions

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/DeviceChannelsHelper.java (hunk 5 above)
Comment (harrymiddelburgetall) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/DeviceChannelsHelper.java]: CosemObjectAccessor for the MBUS_CLIENT_SETUP is now passed to the method so objectconfig is only read onxe

**Review by stefanermens — APPROVED** [review]

**Pull request merged**